### ISSUE TITLE
[Feature Request] Hide Panels and slide in/out animations

### ISSUE THREAD

**Issue by JLeybovichLFL** [yasirkula/UnityDynamicPanels#14]
Ability to hide entire Panels and have them slide in and out.

**Comment by yasirkula** [comment]
You can call panel.gameObject.SetActive to show/hide it. Naturally, it doesn't have a slide animation but IMO the slide animation doesn't apply to all projects and should be implemented by the end-users, if desired.

**Comment by pesseba** [comment]
I'm getting problem when hide/show panels. When a panel hides, the DummyPanel and FreeTarget update well, but when panels are activated back, the DummyPanel Rect doesn't resize and the panels are not docked anymore.  The behaviour can be replicated in DemoScene.

https://github.com/user-attachments/assets/9df2aca5-4519-44fb-ae1d-920fd38c26b9

**Comment by yasirkula** [comment]
@pesseba Dummy panel isn't resized because the panel isn't docked anymore. It isn't docked because there is no guarantee that the panel layout will remain the same before the panel will be reactivated. For example, before reactivating the panel, you could dock the red/green/blue panels to that canvas and now the canvas is not in the same state that it was before.

**Comment by pesseba** [comment]
@yasirkula I did a workaround for my situation saving and restoring the layout. Because sometimes I need to hide the dynamic panels.

**Comment by yasirkula** [comment]
Glad to hear it! Though what happens when you change other panels' positions before restoring the layout? Is only the closed panel's layout restored (somehow)?

**Comment by pesseba** [comment]
I save the layout before hide them. So, after I reactivate the panels first and than I load the layout. All panels works fine. 

https://github.com/user-attachments/assets/428e2475-c0c8-498f-881a-ae9f7408c479

**Comment by yasirkula** [comment]
I meant, what if you moved green panel to the right canvas?

**Comment by yasirkula** [comment]
BTW if you register to PanelNotificationCenter.OnTabClosed or OnPanelClosed, there will be actual close buttons next to tabs and/or panels.

**Comment by pesseba** [comment]
> I meant, what if you moved green panel to the right canvas?

Yes, I can use the panel green from CanvasLeft too. It works fine too.
